### PR TITLE
EWS Extension: enhancements

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/32224

## Description
Added the following commands:
- **ews-get-mailbox-audit-bypass-association**
- **ews-get-user**
- **ews-get-remote-domain**
- **ews-get-federation-configuration**
- **ews-get-federation-trust**

## Minimum version of Demisto
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 1]
```diff
@@ -1008,6 +1364,24 @@ function Main {
             "$script:COMMAND_PREFIX-message-trace-get" {
                 ($human_readable, $entry_context, $raw_response) = GetMessageTraceCommand $exo_client $command_arguments
             }
+            "$script:COMMAND_PREFIX-get-federation-trust" {
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:1367]: move action to be at the end of the command name

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 2]
```diff
@@ -1008,6 +1364,24 @@ function Main {
             "$script:COMMAND_PREFIX-message-trace-get" {
                 ($human_readable, $entry_context, $raw_response) = GetMessageTraceCommand $exo_client $command_arguments
             }
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:1366]: silent warnings

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 3]
```diff
@@ -1008,6 +1364,24 @@ function Main {
...
+            "$script:COMMAND_PREFIX-get-federation-trust" {
+                ($human_readable, $entry_context, $raw_response) = GetFederationTrustCommand $exo_client $command_arguments
+            }
+            "$script:COMMAND_PREFIX-get-federation-configuration" {
+                ($human_readable, $entry_context, $raw_response) = GetFederationConfigurationCommand $exo_client $command_arguments
+            }
+            "$script:COMMAND_PREFIX-get-remote-domain" {
+                ($human_readable, $entry_context, $raw_response) = GetRemoteDomainCommand $exo_client $command_arguments
+            }
+            "$script:COMMAND_PREFIX-get-user" {
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:1376]:
```suggestion
            "$script:COMMAND_PREFIX-users-list" {
```

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 4]
```diff
@@ -944,22 +1219,97 @@ function GetMessageTraceCommand([ExchangeOnlineClient]$client, [hashtable]$kwarg
...
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)][ExchangeOnlineClient]$client,
+        [hashtable]$kwargs
+    )
+    $identity = $kwargs.identity
+    $organizational_unit = $kwargs.organizational_unit
+    $limit = ($kwargs.limit -as [int])
+    $raw_response = $client.GetUser($identity, $organizational_unit, $limit)
+    $human_readable = TableToMarkdown $raw_response "Results of $command"
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:1294]: please open enhancement for TableToMarkdown to support headers

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.yml [hunk 5]
```diff
@@ -218,64 +310,1445 @@ script:
...
+        description: ''
+        type: String
+      - contextPath: EWS.User.WindowsLiveID
+        description: ''
+        type: String
+      - contextPath: EWS.User.DirectReports
+        description: ''
+        type: String
+    deprecated: false
+    description: Use the Get-User command to view existing user objects in your organization.
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.yml:1651]: update desc

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.yml [hunk 6]
```diff
@@ -218,64 +310,1445 @@ script:
...
+      - contextPath: EWS.RemoteDomain.WhenCreatedUTC
+        description: ''
+        type: Date
+    deprecated: false
+    description: View the configuration information for the remote domains configured in your organization. This command is available only in the Exchange Online PowerShell V2 module.
+    execution: false
+    name: ews-get-remote-domain
+  - arguments:
+    - default: false
+      description: The Identity parameter specifies the mailbox you want to view.
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.yml:1307]: add examples

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 7]
```diff
@@ -1008,6 +1364,24 @@ function Main {
...
+            "$script:COMMAND_PREFIX-get-federation-configuration" {
+                ($human_readable, $entry_context, $raw_response) = GetFederationConfigurationCommand $exo_client $command_arguments
+            }
+            "$script:COMMAND_PREFIX-get-remote-domain" {
+                ($human_readable, $entry_context, $raw_response) = GetRemoteDomainCommand $exo_client $command_arguments
+            }
+            "$script:COMMAND_PREFIX-get-user" {
+                ($human_readable, $entry_context, $raw_response) = GetUserCommand $exo_client $command_arguments
+            }
+            "$script:COMMAND_PREFIX-get-mailbox-audit-bypass-association" {
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:1379]: change to list

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 8]
```diff
@@ -1008,6 +1364,24 @@ function Main {
             "$script:COMMAND_PREFIX-message-trace-get" {
                 ($human_readable, $entry_context, $raw_response) = GetMessageTraceCommand $exo_client $command_arguments
             }
+            "$script:COMMAND_PREFIX-get-federation-trust" {
+                ($human_readable, $entry_context, $raw_response) = GetFederationTrustCommand $exo_client $command_arguments
+            }
+            "$script:COMMAND_PREFIX-get-federation-configuration" {
+                ($human_readable, $entry_context, $raw_response) = GetFederationConfigurationCommand $exo_client $command_arguments
+            }
+            "$script:COMMAND_PREFIX-get-remote-domain" {
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:1373]: change everything to list

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.yml [hunk 9]
```diff
@@ -218,64 +310,1445 @@ script:
...
+      required: false
+      secret: false
+    - default: false
+      description: The Identity argument specifies a federation trust ID. If not specified,
+        the command returns all federation trusts configured for the Exchange organization.
+      isArray: false
+      name: identity
+      required: false
+      secret: false
+    - default: false
```
Comment (Itay4) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.yml:1059]: add `auto: PREDEFINED`

File: Packs/EWS/ReleaseNotes/1_8_0.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+##### O365 - EWS - Extension
+- Added the following commands:
+  - **ews-mailbox-audit-bypass-association-list**
+  - **ews--user-list**
```
Comment (hod-alpert) [inline Packs/EWS/ReleaseNotes/1_8_0.md:6]:
```suggestion
  - **ews-user-list**
```

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 11]
```diff
@@ -944,22 +1219,97 @@ function GetMessageTraceCommand([ExchangeOnlineClient]$client, [hashtable]$kwarg
...
-    {
+    else {
         $human_readable = "**No messages trace found**"
     }
 
     return $human_readable, $entry_context, $raw_response
 }
 
+function GetFederationTrustCommand([ExchangeOnlineClient]$client, [hashtable]$kwargs) {
+    # parse arguments
```
Comment (hod-alpert) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:1238]: Add binding and remove return

File: Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1 [hunk 12]
```diff
@@ -944,22 +1219,97 @@ function GetMessageTraceCommand([ExchangeOnlineClient]$client, [hashtable]$kwarg
...
+        $domain_controller, $identity
+    )
+    $human_readable = TableToMarkdown $raw_response "Results of $command"
+    $entry_context = @{
+        "$script:INTEGRATION_ENTRY_CONTEXT.FederationTrust(obj.ApplicationIdentifier === val.ApplicationIdentifier)" = $raw_response
+    }
+    return Write-Output $human_readable, $entry_context, $raw_response
+}
+function GetFederationConfigurationCommand {
+    param (
```
Comment (hod-alpert) [inline Packs/EWS/Integrations/EwsExtension/EwsExtension.ps1:1251]: Add binding

**Review by hod-alpert — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@Itay4 @hod-alpert @gal001 Doc review completed.

**Pull request merged**